File: LeetCode_14/Solution_ZhongJing.java
package LeetCode_14;

/**
 * 日期：2023-07-02 <br>
 * 作者：仲景
 */
public class Solution_ZhongJing {
    public String longestCommonPrefix(String[] strs) {

        String pre = strs[0];

        for (int i = 1; i < strs.length; i++) {
            // 计算strs[i]和公共前缀的公共前缀
            pre = m(strs[i], pre);
            if(pre.length() == 0)
                return "";
        }

        return pre;
    }

    /**
     求两个字符串的公共前缀，没有返回null
     */
    public String m(String s1, String s2) {
        int index = 0;

        while(index < s1.length() && index < s2.length()) {
            if (s1.charAt(index) == s2.charAt(index))
                index++;
            else
                break;
        }

        return s1.substring(0, index);
    }
}

File: LeetCode_14/Solution_ZhongJing.go
package main

// 日期：2023-07-1769_01
// 作者：仲景
func longestCommonPrefix(strs []string) string {
	prefix := strs[0]
	for i := 1; i < len(strs); i++ {
		prefix = twoStrCommonPrefix(strs[i], prefix)
	}

	return prefix
}

// 求两个字符串的公共前缀
func twoStrCommonPrefix(s1, s2 string) string {
	s1Len, s2Len := len(s1), len(s2)
	if s1Len == 0 || s2Len == 0 {
		return ""
	}

	prefix := ""
	for i := 0; i < s1Len && i < s2Len; i++ {
		if s1[i] != s2[i] {
			break
		}
		prefix += string(s1[i])
	}

	return prefix
}
